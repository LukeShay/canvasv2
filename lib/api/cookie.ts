import { serialize, parse } from 'cookie';
import { NextApiRequest, NextApiResponse } from 'next';

export function setTokenCookies(res: NextApiResponse, authorization: string, refresh: string) {
  const authorizationCookie = serialize('authorization', authorization, {
    maxAge: 86400,
    expires: new Date(Date.now() + 86400 * 1000),
    httpOnly: true,
    secure: process.env.NODE_ENV === 'production',
    path: '/',
    sameSite: process.env.NODE_ENV === 'production' ? 'strict' : 'lax',
  });

  const refreshCookie = serialize('refresh', refresh, {
    maxAge: 604800,
    expires: new Date(Date.now() + 604800 * 1000),
    httpOnly: true,
    secure: process.env.NODE_ENV === 'production',
    path: '/',
    sameSite: process.env.NODE_ENV === 'production' ? 'strict' : 'lax',
  });

  res.setHeader('Set-Cookie', [authorizationCookie, refreshCookie]);
}

export function removeTokenCookies(res: NextApiResponse) {
  const authorizationCookie = serialize('authorization', '', { maxAge: -1, path: '/' });
  const refreshCookie = serialize('refresh', '', { maxAge: -1, path: '/' });

  res.setHeader('Set-Cookie', [authorizationCookie, refreshCookie]);
}

export function parseCookies(req: NextApiRequest) {
  if (req.cookies) return req.cookies;

  const cookie = req.headers?.cookie;
  return parse(cookie || '');
}

export function getAuthorizationCookie(req: NextApiRequest) {
  return parseCookies(req).authorization;
}

export function getRefreshCookie(req: NextApiRequest) {
  return parseCookies(req).refresh;
}
