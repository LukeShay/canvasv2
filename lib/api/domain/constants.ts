export default {
  ASSISTANTS_TABLE: 'Assistants',
  CLASSES_TABLE: 'Classes',
  ORGANIZATION_MEMBERS_TABLE: 'OrganizationMembers',
  ORGANIZATIONS_TABLE: 'Organizations',
  SESSIONS_TABLE: 'Sessions',
  STATES_TABLE: 'States',
  STUDENTS_TABLE: 'Students',
  USERS_TABLE: 'Users',
};
