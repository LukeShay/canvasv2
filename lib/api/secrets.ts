function strOrDefault(str?: string) {
  return str || '';
}

export function constants() {
  return {
    accessKeyId: strOrDefault(process.env.AWS_ACCESS_KEY_ID),
    authorizationExpiresIn: strOrDefault(process.env.AUTHORIZATION_EXPIRES_IN),
    authorizationSecret: strOrDefault(process.env.AUTHORIZATION_SECRET),
    awsRegion: strOrDefault(process.env.AWS_REGION),
    refreshExpiresIn: strOrDefault(process.env.REFRESH_EXPIRES_IN),
    refreshSecret: strOrDefault(process.env.REFRESH_SECRET),
    secretAccessKey: strOrDefault(process.env.AWS_SECRET_ACCESS_KEY),
    development: strOrDefault(process.env.NODE_ENV) === 'development',
  };
}
