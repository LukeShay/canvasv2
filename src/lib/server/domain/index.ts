import { Model } from 'objection';
import Knex from 'knex';

export function connectKnex(config: string | Knex.Config): Knex {
  const knex = Knex(config);
  Model.knex(knex);
  return knex;
}

export async function destroyKnex(knex: Knex): Promise<void> {
  await knex.destroy();
}

export * as Constants from './constants';
export * from './models/assistant-model';
export * from './models/class-model';
export * from './models/organization-member-model';
export * from './models/organization-model';
export * from './models/session-model';
export * from './models/state-model';
export * from './models/student-model';
export * from './models/user-model';

export * from './classes/user';
