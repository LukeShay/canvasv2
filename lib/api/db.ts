import { connectKnex } from './domain';

export function connect() {
  if (process.env.NODE_ENV === 'development') {
    connectKnex({
      client: 'sqlite3',
      connection: { filename: '../../dev.sqlite3' },
      useNullAsDefault: true,
    });
  }
}
