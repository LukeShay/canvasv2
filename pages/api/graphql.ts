import { ApolloServer } from 'apollo-server-micro';
import schema from '../../lib/graphql/schema';
import { context, formatError } from '../../lib/graphql/apollo';
import { connect } from '../../lib/api/db';

connect();

const apolloServer = new ApolloServer({
  cacheControl: true,
  context,
  formatError,
  // introspection: constants().development,
  playground: true,
  schema,
  tracing: true,
});

export const config = {
  api: {
    bodyParser: false,
  },
};

export default apolloServer.createHandler({ path: '/api/graphql' });
