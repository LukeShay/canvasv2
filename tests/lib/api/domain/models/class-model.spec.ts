import { ClassModel } from '../../../../../lib/api/domain';

describe('class model', () => {
  it("should have idColum 'id'", () => {
    expect.hasAssertions();
    expect(ClassModel.idColumn).toStrictEqual('id');
  });

  it('should should have jsonSchema that matches snapshot', () => {
    expect.hasAssertions();
    expect(ClassModel.jsonSchema).toMatchSnapshot();
  });

  it('should should have relationMappings that matches snapshot', () => {
    expect.hasAssertions();
    expect(ClassModel.relationMappings).toMatchSnapshot();
  });
});
