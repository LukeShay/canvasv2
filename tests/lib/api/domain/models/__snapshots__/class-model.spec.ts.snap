// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class model should should have jsonSchema that matches snapshot 1`] = `
Object {
  "properties": Object {
    "adminId": Object {
      "type": "string",
    },
    "building": Object {
      "type": Array [
        "string",
        "null",
      ],
    },
    "code": Object {
      "type": "string",
    },
    "description": Object {
      "type": Array [
        "string",
        "null",
      ],
    },
    "name": Object {
      "type": "string",
    },
    "room": Object {
      "type": Array [
        "string",
        "null",
      ],
    },
  },
  "required": Array [
    "adminId",
    "name",
    "code",
  ],
  "type": "object",
}
`;

exports[`class model should should have relationMappings that matches snapshot 1`] = `
Object {
  "admin": Object {
    "join": Object {
      "from": "Classes.adminId",
      "to": "Users.id",
    },
    "modelClass": [Function],
    "relation": [Function],
  },
  "assistants": Object {
    "join": Object {
      "from": "Classes.id",
      "through": Object {
        "from": "Assistants.classId",
        "to": "Assistants.assistantId",
      },
      "to": "Users.Users",
    },
    "modelClass": [Function],
    "relation": [Function],
  },
  "students": Object {
    "join": Object {
      "from": "Classes.id",
      "through": Object {
        "from": "Students.classId",
        "to": "Students.studentId",
      },
      "to": "Users.Users",
    },
    "modelClass": [Function],
    "relation": [Function],
  },
}
`;
